frontend/src/Calendar.js

```diff
@@ -2,19 +2,108 @@ import React from 'react'
 import FullCalendar from '@fullcalendar/react'
 import dayGridPlugin from '@fullcalendar/daygrid'
 
+import moment from 'moment'
+import timeGridPlugin from '@fullcalendar/timegrid'
+
+import {fetchUserCalendar} from './script'
+
+// the conversion of one minute to 60000 milliseconds
+const MIN_TO_MS = 60000;
+
 export default class Calendar extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      fetchDone: false,
```

Comment:
Instead of tracking whether the fetch completed in the state, you could check to see if the current schedule index is `< schedulesTimes.length`. It would mean you have less information to keep track of and your render method won't break if you happen to fetch zero calendars from the backend.

Reply:
This is a better approach. I changed it to `this.state.selectedScheduleId < this.state.schedulesTimes.length`